src/category_theory/adjunction/fully_faithful.lean

```diff
@@ -63,11 +64,60 @@ instance counit_is_iso_of_R_fully_faithful [full R] [faithful R] : is_iso (adjun
     simp,
   end⟩⟩⟩
 
--- TODO also prove the converses?
--- def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L := sorry
--- def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L := sorry
--- def R_full_of_counit_is_iso [is_iso (adjunction.counit h)] : full R := sorry
--- def R_faithful_of_counit_is_iso [is_iso (adjunction.counit h)] : faithful R := sorry
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simp]
+lemma inv_map_unit {X : C} [is_iso (h.unit.app X)] :
+  inv (L.map (h.unit.app X)) = h.counit.app (L.obj X) :=
+is_iso.inv_eq_of_hom_inv_id (h.left_triangle_components)
+
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simps]
+noncomputable def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] :
+  L ⋙ R ⋙ L ≅ L :=
+(functor.associator _ _ _).symm ≪≫
```

Comment:
Here and in the dual this can be shortened a little (to make it fit on one line), eg `(functor.associator _ _ _).symm` is actually saying `(functor.associator L R L).symm`, but now we can use dot-notation and write this as `(L.associator R L).symm`. Similarly `iso_whisker_right (as_iso h.unit).symm L` works too! You don't need to make these changes, but I thought you might like to be aware of them :)